### PR TITLE
[FIX] stock: toggle unreserve button correctly

### PR DESCRIPTION
**Pull request by svs-odoo**
Before this commit, a picking can be in a state where it has all its moves marked as assigned but with no reserved quantity.

How to reproduce:
- Create a product tracked by lot;
- Use inventory adjustment to create two lots for this product:
    - one with a quantity;
    - one without;
- Create a delivery, add a line for this product;
- Confirm the picking and click on "Check Availability";
- The move line must have the lot with quantity and must have reserved quantity.
- Change the lot id of the move line.
    --> Move line doesn't have reserved quantity anymore but the move is still assigned and the "Check Availability" button isn't displayed but the "Unreserve" is.

task-2171546

### CODE REVIEW

File: addons/stock/models/stock_move_line.py [hunk 1]
```diff
@@ -273,6 +273,14 @@ def write(self, vals):
                     if new_product_qty != ml.product_qty:
                         new_product_uom_qty = ml.product_id.uom_id._compute_quantity(new_product_qty, ml.product_uom_id, rounding_method='HALF-UP')
                         ml.with_context(bypass_reservation_update=True).product_uom_qty = new_product_uom_qty
+                        for move in ml.mapped('move_id'):
```
Comment (fmdl) [inline addons/stock/models/stock_move_line.py]: `for move in ml.mapped('move_id').filtered(lambda x: x.state == 'assigned'):`
remove on tab :-)

File: addons/stock/models/stock_move_line.py [hunk 2]
```diff
@@ -273,6 +273,14 @@ def write(self, vals):
...
+                            if move.state == 'assigned':
+                                qty = sum(move.move_line_ids.mapped('product_uom_qty'))
+                                if move.reserved_availability != qty:
```
Comment (fmdl) [inline addons/stock/models/stock_move_line.py]: use `float_comparaison()`

File: addons/stock/models/stock_move_line.py [hunk 3]
```diff
@@ -273,6 +273,14 @@ def write(self, vals):
...
+                                    if qty == 0:
```
Comment (fmdl) [inline addons/stock/models/stock_move_line.py]: use `float_zero()`

**Comment by C3POdoo** [comment]
Dear @svs-odoo,

Thank you for your contribution but the version 12.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**